packages/jest-haste-map/src/index.ts

```diff
@@ -254,23 +254,11 @@ class HasteMap extends EventEmitter {
     };
     this._console = options.console || global.console;
 
-    if (options.ignorePattern) {
-      if (options.ignorePattern instanceof RegExp) {
-        this._options.ignorePattern = new RegExp(
-          options.ignorePattern.source.concat('|' + VCS_DIRECTORIES),
-          options.ignorePattern.flags,
-        );
-      } else {
-        const ignorePattern = options.ignorePattern;
-        const vcsIgnoreRegExp = new RegExp(VCS_DIRECTORIES);
-        this._options.ignorePattern = (filePath: string) =>
-          vcsIgnoreRegExp.test(filePath) || ignorePattern(filePath);
-
-        this._console.warn(
-          'jest-haste-map: the `ignorePattern` options as a function is being ' +
-            'deprecated. Provide a RegExp instead. See https://github.com/facebook/jest/pull/4063.',
-        );
-      }
+    if (options.ignorePattern && options.ignorePattern instanceof RegExp) {
+      this._options.ignorePattern = new RegExp(
+        options.ignorePattern.source.concat('|' + VCS_DIRECTORIES),
+        options.ignorePattern.flags,
+      );
```

Comment:
```suggestion
    if (options.ignorePattern) {
      if (options.ignorePattern instanceof RegExp) `{
        this._options.ignorePattern = new RegExp(
          options.ignorePattern.source.concat('|' + VCS_DIRECTORIES),
          options.ignorePattern.flags,
        );
      } else {
        throw new Error('jest-haste-map: the `ignorePattern` option must be a RegExp');
      }
```

Just so we don't silently ignore it if it's the wrong type 🙂  Can make the test you deleted jut assert it throws

Reply:
Thanks. I think so too. This code is testable and friendly. 👍